dbms/src/Common/TiFlashSecurity.h

```diff
@@ -32,41 +36,76 @@ extern const int INVALID_CONFIG_PARAMETER;
 
 struct TiFlashSecurityConfig
 {
-    String ca_path;
-    String cert_path;
-    String key_path;
+public:
+    TiFlashSecurityConfig() = default;
 
-    bool redact_info_log = false;
+    TiFlashSecurityConfig(Poco::Util::AbstractConfiguration & config, const LoggerPtr & log_)
+        : log(log_)
+    {
+        update(config);
+    }
 
-    std::set<String> allowed_common_names;
+    void setLog(const LoggerPtr & log_)
+    {
+        std::unique_lock lock(mu);
+        log = log_;
+    }
 
-    bool inited = false;
-    bool has_tls_config = false;
-    grpc::SslCredentialsOptions options;
+    bool hasTlsConfig()
+    {
+        std::unique_lock lock(mu);
+        return has_tls_config;
+    }
 
-public:
-    TiFlashSecurityConfig() = default;
+    bool redactInfoLog()
+    {
+        std::unique_lock lock(mu);
+        return redact_info_log;
+    }
+
+    std::tuple<String, String, String> getPaths()
+    {
+        std::unique_lock lock(mu);
+        return {ca_path, cert_path, key_path};
+    }
 
-    TiFlashSecurityConfig(Poco::Util::LayeredConfiguration & config, const LoggerPtr & log)
+    std::pair<String, String> getCertAndKeyPath()
     {
+        std::unique_lock lock(mu);
+        return {cert_path, key_path};
+    }
+
+    std::set<String> allowedCommonNames()
+    {
+        std::unique_lock lock(mu);
+        return allowed_common_names;
+    }
+
+    bool update(Poco::Util::AbstractConfiguration & config)
+    {
+        std::unique_lock lock(mu);
         if (config.has("security"))
```

Comment:
This should not happen. If we config the cluster with security config, we will not remove it online, since the flashService or http server will always use sslServerCredential.

Reply:
So my understand is the security config should not be changed online, neither from non-security to security nor from security to non-security? If yes, I think we should check this and reject this kind of change explicitly?

Reply:
> So my understand is the security config should not be changed online, neither from non-security to security nor from security to non-security? If yes, I think we should check this and reject this kind of change explicitly?

Yes, I will handle it